utils/__tests__/profile.test.ts

```diff
@@ -54,3 +90,165 @@ describe("retrievedProfileToLimitedProfile", () => {
     );
   });
 });
+
+const aRetrievedProfileWithLegacyPreferences = {
+  ...aRetrievedProfile,
+  servicePreferencesSettings: legacyProfileServicePreferencesSettings
+};
+
+const aRetrievedProfileWithManualPreferences = {
+  ...aRetrievedProfile,
+  servicePreferencesSettings: manualProfileServicePreferencesSettings
+};
+
+const aRetrievedProfileWithAutoPreferences = {
+  ...aRetrievedProfile,
+  servicePreferencesSettings: autoProfileServicePreferencesSettings
+};
+
+const mockProfileFindLast = jest.fn(() =>
+  taskEither.of(some(aRetrievedProfileWithLegacyPreferences))
+);
+const mockProfileModel = ({
+  findLastVersionByModelId: mockProfileFindLast
+} as unknown) as ProfileModel;
+
+const mockServicePreferenceFind = jest.fn();
+const mockServicePreferenceModel = ({
+  find: mockServicePreferenceFind
+} as unknown) as ServicesPreferencesModel;
+
+// utility that adds a given set of serviceIds to the profile's inbox blacklist
+const withBlacklist = (profile: RetrievedProfile, services = []) =>
+  services.reduce((p, serviceId) => {
+    const { blockedInboxOrChannels = {}, ...restOfProfile } = p;
+    return {
+      ...restOfProfile,
+      blockedInboxOrChannels: {
+        ...blockedInboxOrChannels,
+        [serviceId]: [BlockedInboxOrChannelEnum.INBOX]
+      }
+    };
+  }, profile);
+
```

Comment:
Another way without recreating profile 
```suggestion
// utility that adds a given set of serviceIds to the profile's inbox blacklist
const withBlacklist = (profile: RetrievedProfile, services = []) => {
  return {
    ...profile,
    blockedInboxOrChannels: services.reduce(
      (obj, serviceId) => ({
        ...obj,
        [serviceId]: BlockedInboxOrChannelEnum.INBOX
      }),
      {}
    )
  };
```